docs/sc-memory/api/cpp/extended/permissions_api.md

```diff
@@ -0,0 +1,809 @@
+      <td><code>action_read_from_sc_memory</code></td>
+      <td>R</td>
+      <td>The user can execute all methods that read sc-constructions from knowledge base.</td>
+    </tr>
+    <tr>
+      <td>generating in sc-memory action</td>
+      <td><code>action_generate_in_sc_memory</code></td>
+      <td>G</td>
+      <td>The user can execute all methods that generate sc-constructions in knowledge base.</td>
+    </tr>
+    <tr>
+      <td>erasing from sc-memory action</td>
+      <td><code>action_erase_from_sc_memory</code></td>
+      <td>E</td>
+      <td>The user can execute all methods that erase sc-constructions from knowledge base.</td>
+    </tr>
+    <tr>
+      <td>reading permissions from sc-memory action</td>
+      <td><code>action_read_permissions_from_sc_memory</code></td>
+      <td>RP</td>
+      <td>The user can read permissions of other users from knowledge base.</td>
+    </tr>
+    <tr>
+      <td>generating permissions in sc-memory action</td>
+      <td><code>action_generate_permissions_in_sc_memory</code></td>
+      <td>GP</td>
+      <td>The user can generate permissions for other users in knowledge base.</td>
+    </tr>
+    <tr>
+      <td>erasing permissions from sc-memory action</td>
+      <td><code>action_erase_permissions_from_sc_memory</code></td>
+      <td>EP</td>
+      <td>The user can erase permissions of other users from knowledge base.</td>
+    </tr>
+  </tbody>
+</table>
+
+Each core method of [**C++ Core API**](../core/api.md) checks user permissions. All methods of *C++ Extended API* use methods of *C++ Core API*. Description of ScMemoryContext API methods and user permissions required to execute these methods is represented below:
+
+<table>
+  <thead>
+    <tr>
+      <th>ScMemoryContext API method</th>
+      <th>R</th>
+      <th>G</th>
+      <th>E</th>
+      <th>Logic if user hasn't required permissions to perform method</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>CreateNode, CreateLink</td>
+      <td>-</td>
+      <td>-</td>
+      <td>-</td>
+      <td>These methods don't require any permissions for users.</td>
+    </tr>
+    <tr>
+      <td>CreateEdge</td>
+      <td>-</td>
+      <td>+</td>
+      <td>-</td>
+      <td>If user hasn't permissions to add sc-connector from specified begin sc-element or to specified end sc-element, then method will throw <code>utils::ExceptionInvalidState</code>.</td>
+    </tr>
+    <tr>
+      <td>IsElement</td>
+      <td>+</td>
+      <td>-</td>
+      <td>-</td>
+      <td>If user hasn't permissions to read specified sc-element, then method will throw <code>utils::ExceptionInvalidState</code>.</td>
+    </tr>
+    <tr>
+      <td>GetElementType</td>
+      <td>+</td>
+      <td>-</td>
+      <td>-</td>
+      <td>If user hasn't permissions to read specified sc-element, then method will throw <code>utils::ExceptionInvalidState</code>.</td>
+    </tr>
+    <tr>
+      <td>SetElementSubtype</td>
+      <td>-</td>
```

Comment:
any set  and erase operation requires read permissions, let's discuss this

Reply:
Ok, lets discuss